test/integration/install_test.go

```diff
@@ -404,6 +404,15 @@ func TestInstallOrUpgradeCli(t *testing.T) {
 			"'kubectl apply' command failed\n%s", out)
 	}
 
+	// Wait for the proxy injector to be up
+	name := "linkerd-proxy-injector"
+	ns := "linkerd"
+	o, err := TestHelper.Kubectl("", "--namespace="+ns, "rollout", "status", "--timeout=120s", "deploy/"+name)
+	if err != nil {
+		testutil.AnnotatedFatalf(t, fmt.Sprintf("failed to wait for condition=available for deploy/%s in namespace %s", name, ns),
+			"failed to wait for condition=available for deploy/%s in namespace %s: %s: %s", name, ns, err, o)
+	}
+
 	// Install Linkerd Viz Extension
```

Comment:
Unrelated Issue: We currently don;t run the upgrade tests for viz correctly i.e staring with the stable/edge install and then installing the source. We only do source install currently

I will raise an another PR to do that